### DISCUSSION TITLE
Addition - Filament sensor connection

### DISCUSSION THREAD

**Discussion by soosp** [1coderookie/Kobra2NeoInsights#11] [General]
I bougth a filament sensor for my Kobra 2 Neo from [Aliexpress](https://www.aliexpress.com/item/1005006528910107.html).

![filament_sensor_1](https://github.com/user-attachments/assets/632a0e06-906e-47ed-9ce7-da608878976f)

I had to rewirig the connector labeled E to make it works. You have to connect wires to the first and last position of the connector.

![filament_sensor_2](https://github.com/user-attachments/assets/6261cddb-5c4d-4f45-8762-7916f798071a)

![filament_sensor_3](https://github.com/user-attachments/assets/396cd0fa-07c6-4ed4-b507-3d6f3a9d843a)

These pins are in closed position if filament is out.  If the filament is loaded  the sensor contact is opened. If the contact is closed because the filament is out, the printer is pausing. If you disconnect the sensor, the wires are opened as if the filament is loaded and printer is working. So if you leave wires at factory position it will works reverse way (at least with this sensor).

**Comment by 1coderookie** [comment]
Thanks! 
I'm not sure if all those switches are the same, but at least at the stock sensor we have three pins there at the actual switch (GND, NO, NC) and the wires are connected to GND and NO. IIrc the switch is NO with filament OUT, but I'll check that again.
